borg/helpers.py

```diff
@@ -279,15 +279,32 @@ def list_archive_infos(self, sort_by=None, reverse=False):
         return archives
 
 
-def prune_within(archives, within):
+def interval(s):
+    """Convert a string representing a valid interval to a number of hours."""
     multiplier = {'H': 1, 'd': 24, 'w': 24 * 7, 'm': 24 * 31, 'y': 24 * 365}
+
+    if s.endswith(tuple(multiplier.keys())):
+        number = s[:-1]
+        suffix = s[-1]
+    else:
+        # range suffixes in ascending multiplier order
+        ranges = [k for k, v in sorted(multiplier.items(), key=lambda t: t[1])]
+        raise argparse.ArgumentTypeError(
+            'Unexpected interval time unit "%s": expected one of %s' % (s[-1], ranges))
```

Comment:
I think the repr-like transform leads to a more clear error message.  I'll change the placeholder.